x/ccv/provider/keeper/distribution.go

```diff
@@ -240,6 +261,12 @@ func (k Keeper) GetConsumerRewardsPool(ctx sdk.Context) sdk.Coins {
 func (k Keeper) ComputeConsumerTotalVotingPower(ctx sdk.Context, chainID string) (totalPower int64) {
 	// sum the opted-in validators set voting powers
 	for _, v := range k.GetConsumerValSet(ctx, chainID) {
+
+		// only consider the voting power of a validator that would receive rewards (i.e., validator has been validating for a number of blocks)
+		if !k.IsEligibleForConsumerRewards(ctx, v.Height) {
+			continue
+		}
```

Comment:
Does this mean that a validator not being eligible results in the other validators getting more rewards?

Reply:
Yes, this could potentially happen. For instance, consider you have 2 validators `A` and `B` with powers 100 and 400 respectively, so the `totalPower` would have been 500. If both validators are eligible, then `A` would get `100/500` of the tokens, while `B` would get `400 / 500` of the tokens. However, if validator `B` is not eligible for rewards, this would mean that `totalPower` is only `100` and hence `A` gets all the tokens.